src/components/avatar/avatar.test.ts

```diff
@@ -98,13 +96,13 @@ describe('<sl-avatar>', () => {
       await expect(el).to.be.accessible();
     });
 
-    it('should accept as an assigned child in the shadow root', async () => {
-      const slot = <HTMLSlotElement>el.shadowRoot.querySelector('slot[name=icon]');
-      const childNodes = slot.assignedNodes({ flatten: true });
+    it('should accept as an assigned child in the shadow root', () => {
+      const slot = el.shadowRoot!.querySelector<HTMLSlotElement>('slot[name=icon]')!;
```

Comment:
For the record, it already is pretty smart. It can already determine basic selectors. [See playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBBA2JYF4YBMTAK4FsCmYUAdAI5Z4BOAngMp7x7BQgUAUA5AkuwJQCEAbgCwAKC7EAhhAgBLAOZg8aAHIg0eCKx7CRo0JFhgkNRCnSZcBYmUq16jZm3YAjLFGZheg0UagmkRFKyCkqq6praQA)